Source/GSStoryboardTransform.m

```diff
@@ -0,0 +1,871 @@
+#import <Foundation/NSException.h>
+
+#import "AppKit/NSSeguePerforming.h"
+#import "AppKit/NSStoryboard.h"
+#import "AppKit/NSStoryboardSegue.h"
+#import "AppKit/NSNibDeclarations.h"
+#import "AppKit/NSViewController.h"
+#import "AppKit/NSWindowController.h"
+
+#import "GSStoryboardTransform.h"
+#import "GSFastEnumeration.h"
+
+#define APPLICATION @"application"
+
+@interface NSStoryboardSegue (__private__)
+- (void) _setDestinationController: (id)controller; 
+- (void) _setSourceController: (id)controller;
+@end
+
+@interface NSStoryboardSegue (__StoryboardPrivate__)
+// Private to this class...
+- (void) _setKind: (NSString *)k;
+- (void) _setRelationship: (NSString *)r;
+- (NSString *) _kind;
+- (NSString *) _relationship;
+@end
+
+// this needs to be set on segues
+@implementation NSStoryboardSegue (__StoryboardPrivate__)
+- (void) _setKind: (NSString *)k
+{
+  ASSIGN(_kind, k);
+}
+
+- (void) _setRelationship: (NSString *)r
+{
+  ASSIGN(_relationship, r);
+}
+
+- (NSString *) _kind
+{
+  return _kind;
+}
+
+- (NSString *) _relationship
+{
+  return _relationship;
+}
+@end
+
+@implementation NSStoryboardSeguePerformAction
+- (id) target
+{
+  return _target;
+}
+
+- (void) setTarget: (id)target
+{
+  ASSIGN(_target, target);
+}
+
+- (SEL) action
+{
+  return _action;
+}
+
+- (void) setAction: (SEL)action
+{
+  _action = action;
+}
+
+- (NSString *) selector
+{
+  return NSStringFromSelector(_action);
+}
+
+- (void) setSelector: (NSString *)s
+{
+  _action = NSSelectorFromString(s);
+}
+
+- (id) sender
+{
+  return _sender;
+}
+
+- (void) setSender: (id)sender
+{
+  ASSIGN(_sender, sender);
+}
+
+- (NSString *) identifier
+{
+  return _identifier;
+}
+
+- (void) setIdentifier: (NSString *)identifier
+{
+  ASSIGN(_identifier, identifier);
+}
+
+- (NSString *) kind
+{
+  return _kind;
+}
+
+- (void) setKind: (NSString *)kind
+{
+  ASSIGN(_kind, kind);
+}
+
+- (NSStoryboard *) storyboard
+{
+  return _storyboard;
+}
+
+- (void) setStoryboard: (NSStoryboard *)storyboard
+{
+  ASSIGN(_storyboard, storyboard);
+}
+
+- (NSStoryboardSegue *) storyboardSegue
+{
+  return _storyboardSegue;
+}
+
+- (void) setStoryboardSegue: (NSStoryboardSegue *)ss
+{
+  ASSIGN(_storyboardSegue, ss);
+}
+
+- (void) dealloc
+{
+  RELEASE(_storyboard);
+  RELEASE(_kind);
+  RELEASE(_identifier);
+  RELEASE(_sender);
+  RELEASE(_storyboardSegue);
+  [super dealloc];
+}
+
+- (IBAction) doAction: (id)sender
+{
+  BOOL should = YES;
+
+  // If the instance we are testing is a controller, then the value of should is set by this method....
+  // if it is not, as it is possible to initiate a segue from an NSMenuItem, then we don't, but should
+  // remains set to YES so that the logic to replace the destination controller is still called.
+  if ([_sender respondsToSelector: @selector(shouldPerformSegueWithIdentifier:sender:)]  &&
+      ([_sender isKindOfClass: [NSViewController class]] || [_sender isKindOfClass: [NSWindowController class]]))
```

Comment:
I think these extra test here are not required. Rather you should add a test further down before calling performSegueWithIdentifier:sender:

Reply:
What test specifically?

Reply:
Added a check to see if the _sender responds to the performSegueForIdentifier:... method.